code/mob/suicide.dm

```diff
@@ -180,14 +180,35 @@
 	if (.)
 		src.visible_message("<span class='alert'><b>[src] suddenly dies for no adequately explained reason!</b></span>")
 
-// instead of dying, flockdrone suicide should hand control back to the mobcritter AI
 /mob/living/critter/flock/drone/do_suicide()
-	emote("beep")
-	say("\[System notification: self-aware drone has voluntarily surrendered its self-awareness and returned to basic function.\]")
-	var/mob/living/C = src.controller
-	src.release_control()
-	if(C)
-		C.suicide()
-		C.unlock_medal("Damned", 1)
-
-	return TRUE
+	if ((locate(/obj/flock_structure/relay) in src.flock.structures) && istype(src.controller, /mob/living/intangible/flock/flockmind) && !length(src.flock.getActiveTraces()))
+		boutput(src, "<span class='alert'>You can't abandon your Flock with the Relay active!</span>")
+		return
+	if (tgui_alert(src, "Are you sure you want to commit suicide?", "Confirm Suicide", list("Yes", "No")) == "Yes")
+		var/mob/living/intangible/flock/C = src.controller
+		src.release_control()
+		if (istype(C, /mob/living/intangible/flock/flockmind))
+			var/mob/living/intangible/flock/flockmind/F = C
+			F.do_suicide(TRUE)
+		else
+			var/mob/living/intangible/flock/trace/T = C
+			T.do_suicide(TRUE)
+
+/mob/living/intangible/flock/flockmind/do_suicide(skip_prompt)
+	if ((locate(/obj/flock_structure/relay) in src.flock.structures) && !length(src.flock.getActiveTraces()))
+		boutput(src, "<span class='alert'>You can't abandon your Flock with the Relay active!</span>")
+		return
+	if (skip_prompt || tgui_alert(src, "Are you sure you want to commit suicide?", "Confirm Suicide", list("Yes", "No")) == "Yes")
+		var/result = src.getTraceToPromote()
```

Comment:
What does this return if the flockmind that is suiciding just closes the game or something instead of picking from the list?

Reply:
Nothing happens when closing the game, so would think it just cancels the prompt

Reply:
Maybe pick a trace at random if the prompt is cancelled?

Reply:
Idk, would say not necessary, since still some time to back out if you want I suppose. If done to prevent just quitting the game, will need some afk timer check in the future for quitting without suiciding, so could probably just use that instead?

Reply:
hmm, maybe you're right. I guess leave this for now, and if it actually becomes an issue in the future, we can fix it then.

Reply:
Hm?

Reply:
I'm saying that cancel should just close the prompt, not do something else.